src/zenml/integrations/mlflow/steps/mlflow_registry.py

```diff
@@ -0,0 +1,168 @@
+        trained_model_name: Name of the model to be deployed.
+        experiment_name: Name of the experiment to be used for the run.
+        run_name: Name of the run to be created.
+        run_id: ID of the run to be used.
+        model_source_uri: URI of the model source. If not provided, the model
+            will be fetched from the MLflow tracking server.
+        version_description: Description of the model.
+        version_tags: Tags to be added to the model.
+        registry_metadata: Metadata to be added to the model registry.
+    """
+
+    name: str
+    description: Optional[str] = None
+    tags: Optional[Dict[str, str]] = None
+    trained_model_name: Optional[str] = "model"
+    model_source_uri: Optional[str] = None
+    experiment_name: Optional[str] = None
+    run_name: Optional[str] = None
+    run_id: Optional[str] = None
+    version_description: Optional[str] = None
+    version_tags: Optional[Dict[str, str]] = None
+    registry_metadata: Optional[Dict[str, str]] = None
+
+
+@step(enable_cache=True)
+def mlflow_register_model_step(
+    model: UnmaterializedArtifact,
+    params: MLFlowRegistryParameters,
+) -> None:
+    """MLflow model registry step.
+
+    Args:
+        model: Model to be registered, This is not used in the step, but is
+            required to trigger the step when the model is trained.
+        params: Parameters for the step.
+
+    Raises:
+        ValueError: If the model registry is not an MLflow model registry.
+        get_missing_mlflow_experiment_tracker_error: If the experiment tracker
+            is not an MLflow experiment tracker.
+        ValueError: If no model source URI is provided and no model is found
+    """
+    # get the experiment tracker and check if it is an MLflow experiment tracker.
+    experiment_tracker = Client().active_stack.experiment_tracker
+    if not isinstance(experiment_tracker, MLFlowExperimentTracker):
+        raise get_missing_mlflow_experiment_tracker_error()
+
+    # fetch the MLflow model registry
+    model_registry = Client().active_stack.model_registry
+    if not isinstance(model_registry, MLFlowModelRegistry):
+        raise ValueError(
+            "The MLflow model registry step can only be used with an "
+            "MLflow model registry."
+        )
+
+    # get pipeline name, step name and run id
+    step_env = cast(StepEnvironment, Environment()[STEP_ENVIRONMENT_NAME])
+    pipeline_name = step_env.pipeline_name
+    step_name = step_env.step_name
+    pipeline_run_id = step_env.run_name
+
+    # Get MLflow run ID either from params or from experiment tracker using
+    # pipeline name and run name
+    mlflow_run_id = params.run_id or experiment_tracker.get_run_id(
+        experiment_name=params.experiment_name or pipeline_name,
+        run_name=params.run_name or pipeline_run_id,
+    )
+    # If no value was set at all, raise an error
+    if not mlflow_run_id:
+        raise ValueError(
```

Comment:
I think most of the `ValueError`s you raise here should be `RuntimeError`s instead. ValueErrors are raised when the input to a function does not have the correct value, e.g.
```python
def divide(numerator: int, denominator: int):
  if denominator == 0:
    raise ValueError("Can't divide by 0")
  return numerator/denominator
```
```
https://docs.python.org/3/library/exceptions.html#ValueError